app/services/proxy_status/base.rb

```diff
@@ -0,0 +1,40 @@
+module ProxyStatus
+  class Base
+    def initialize(proxy, opts = {}, api_class_name = 'Version')
+      @proxy = proxy
+      @cache_duration = opts[:cache_duration] || 3.minutes
+      @api = "ProxyAPI::#{api_class_name}".classify.constantize.new(:url => proxy.url)
+    end
+
+    def api_versions
+      Rails.cache.fetch(cache_key, :expires_in => cache_duration) do
+        fetch_proxy_data do
+          api.proxy_versions
+        end
+      end
+    end
+
+    def revoke_cache!
+      # As memcached does not support delete_matched, we need to delete each
+      Rails.cache.delete(cache_key)
+    end
+
+    def cache_key
+      "proxy_#{proxy.id}/versions"
+    end
+
+    private
+
+    attr_reader :proxy, :cache_duration, :api
+    alias_method :version, :api_versions
+
+    def fetch_proxy_data
+      begin
+        yield
+      rescue *ProxyStatus::CONNECTION_ERRORS => exception
+        raise ::Foreman::WrappedException.new exception, N_("Unable to connect to smart proxy")
```

Comment:
I would consider changing this just to "unable to connect", imho its clear you are talking about a proxy.